packages/@aws-cdk/aws-cloudfront/lib/origin.ts

```diff
@@ -167,6 +167,7 @@ export abstract class OriginBase implements IOrigin {
 
   private renderCustomHeaders(): CfnDistribution.OriginCustomHeaderProperty[] | undefined {
     if (!this.customHeaders || Object.entries(this.customHeaders).length === 0) { return undefined; }
+    validateCustomHeaders(this.customHeaders);
```

Comment:
`~` I have a slight preference for moving this up to the constructor, with the other `validate*` calls. This would also have the bonus of making the s3-origin tests slightly simpler (only need to construct the S3Origin, not bind it to a Distribution).

Reply:
Done